libs/data/bbhnet/data/utils.py

```diff
@@ -66,14 +69,31 @@ def sample_kernels(
     # always make sure that the center of x's
     # 1st axis is in the kernel that we sample
     # if we're doing >1D sampling
-    min_sample_start = max(
-        x.shape[-1] // 2 + 1 - trigger_distance_size - (size / 2), 0
-    )
-    max_sample_start = min(
-        x.shape[-1] // 2 - 1 - (size / 2) + trigger_distance_size,
-        x.shape[-1] - size,
-    )
 
+    # TODO: not sure if theres a logical way to

```

Comment:
I don't think there is, or if there is I'm not sure it would be more clear than this, so I think this makes sense. The one thing I might recommend is setting something like `dim = x.shape[-1]` to save some keystrokes and hopefully even some lines.

Reply:
Maybe even add in `t0 = dim // 2 + 1` to make things extra clear blow but might be overkill